File: app-foretime/src/main/java/com/smallraw/foretime/app/repository/db/dao/ConfigDao.java
package com.smallraw.foretime.app.repository.db.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.smallraw.foretime.app.repository.db.entity.ConfigDO;

import java.util.List;

@Dao
public interface ConfigDao {
  @Query("SELECT COUNT(*) FROM config")
  int count();

  @Insert
  long insert(ConfigDO configDO);

  @Query("SELECT * FROM config WHERE id = :id")
  ConfigDO findById(long id);

  @Query("SELECT * FROM config WHERE name= :name")
  List<ConfigDO> findByKey(String name);

  @Update
  int update(ConfigDO configDO);
}

File: app-foretime/src/main/java/com/smallraw/foretime/app/repository/db/dao/ConfigDao.kt
package com.smallraw.foretime.app.repository.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import android.arch.persistence.room.Update

import com.smallraw.foretime.app.repository.db.entity.ConfigDO

@Dao
interface ConfigDao {
    @Query("SELECT COUNT(*) FROM config")
    fun count(): Int

    @Insert
    fun insert(configDO: ConfigDO): Long

    @Query("SELECT * FROM config WHERE id = :id")
    fun findById(id: Long): ConfigDO

    @Query("SELECT * FROM config WHERE name= :name")
    fun findByKey(name: String): List<ConfigDO>

    @Update
    fun update(configDO: ConfigDO): Int
}
